service/api/client/client_test.go

```diff
@@ -172,16 +185,67 @@ func TestClient_AddRecord(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	t.Run("test add record", func(t *testing.T) {
-		rec, err := client.AddRecord(context.Background(), info.ID, body)
+	t.Run("test create record", func(t *testing.T) {
+		rec, err := client.CreateRecord(context.Background(), info.ID, body)
 		if err != nil {
-			t.Fatalf("failed to add record: %v", err)
+			t.Fatalf("failed to create record: %v", err)
 		}
 		if !rec.ThreadID().Equals(info.ID) {
-			t.Fatal("got bad thread ID from add record")
+			t.Fatal("got bad thread ID from create record")
 		}
 		if rec.LogID().String() == "" {
-			t.Fatal("got bad log ID from add record")
+			t.Fatal("got bad log ID from create record")
+		}
+	})
+}
+
+func TestClient_AddRecord(t *testing.T) {
+	t.Parallel()
+	_, client, done := setup(t)
+	defer done()
+
+	// Create a thread, keeping read key and log private key on the client
+	id := thread.NewIDV1(thread.Raw, 32)
+	fk, err := symmetric.CreateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+	sk, pk, err := crypto.GenerateEd25519Key(crand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+	info, err := client.CreateThread(context.Background(), id, core.FollowKey(fk), core.LogKey(pk))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	body, err := cbornode.WrapObject(map[string]interface{}{
+		"foo": "bar",
+		"baz": []byte("howdy"),
+	}, mh.SHA2_256, -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("test add record", func(t *testing.T) {
+		rk, err := symmetric.CreateKey()
+		if err != nil {
+			t.Fatal(err)
+		}
+		event, err := cbor.CreateEvent(context.Background(), nil, body, rk)
```

Comment:
The tests use the cbor event and record methods, which take a DAGService, which adds the nodes. The client doesn't care about that since the remote service will do this anyway. So, here we pass nil for the DAGService.

Reply:
Beautiful! Aside from the LogKey, this is quite possibly the best part about this new API. Making explicit that you are _not_ responsible for adding the content to IPFS is really nice here. It also makes extending this to a full peer with a DAG Service _later_ sooo much easier (conceptually).